test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -182,14 +193,78 @@ class db_size_statistic : public statistic {
     int64_t _limit;
 };
 
+class postrun_statistic_check {
+    public:
+    explicit postrun_statistic_check(configuration *config)
+    {
+        const auto config_stats = config->get_list(POSTRUN_STATISTICS);
+        /*
+         * Each stat in the configuration is a colon separated list in the following format:
+         * - Stat name
+         * - Min limit
+         * - Max limit
+         */
+        for (const auto &c : config_stats) {
+            auto stat = split_string(c, ':');
+            if (stat.size() != 3)
+                testutil_die(-1,
+                  "runtime_monitor: Each postrun statistic must follow the format of "
+                  "\"stat_name:min_limit:max_limit\". Invalid format \"%s\" provided.",
+                  c.c_str());
+            const int min_limit = std::stoi(stat.at(1)), max_limit = std::stoi(stat.at(2));
+            if (min_limit > max_limit)
+                testutil_die(-1,
+                  "runtime_monitor: The min limit of each postrun statistic must be less than or "
+                  "equal to its max limit. Config=\"%s\" Min=%ld Max=%ld",
+                  c.c_str(), min_limit, max_limit);
+            _stats.emplace_back(std::move(stat.at(0)), min_limit, max_limit);
+        }
+    }
+    virtual ~postrun_statistic_check() = default;
+
+    void
+    check(scoped_cursor &cursor) const
+    {
+        for (const auto &stat : _stats)
+            check_stat(cursor, stat);
+    }
+
+    private:
+    struct postrun_statistic {
+        postrun_statistic(std::string &&name, const int64_t min_limit, const int64_t max_limit)
+            : name(std::move(name)), field(get_stat_field(this->name)), min_limit(min_limit),
+              max_limit(max_limit)
+        {
+        }
+        const std::string name;
+        const int field;
+        const int64_t min_limit, max_limit;
+    };
+    void
+    check_stat(scoped_cursor &cursor, const postrun_statistic &stat) const
+    {
+        int64_t stat_value;
+
+        testutil_assert(cursor.get() != nullptr);
+        get_stat(cursor, stat.field, &stat_value);
+        if (stat_value < stat.min_limit || stat_value > stat.max_limit)
+            testutil_die(-1,
```

Comment:
See my comment about checking all stats first.
I'd simply print the stat info if it is beyond the limits but not die here. Having all stats that fail could help to understand what's wrong ? Or would it be too many stats ?

Reply:
Good idea! I'll do that.

Reply:
15dcaa3